File: src/main/java/racingcar/Car.java
package racingcar;

public class Car {

    private final String name;
    private int position;

    public Car(String name) {
        if (name.length() > 5) {
            throw new IllegalArgumentException();
        }
        this.name = name;
    }

    public int position() {
        return this.position;
    }

    public void move(int value) {
        if (value >= 4) {
            this.position++;
        }
    }
}

File: src/main/java/racingcar/Car.kt
package racingcar

class Car(
    val name: String,
) {
    var position: Int = 0

    init {
        require(name.length > 5) { throw IllegalArgumentException("자동차의 이름은 5글자를 초과하면 안됩니다.") }
    }

    fun move(movingStartegy: MovingStrategy) {
        if (movingStartegy.movable()) {
            position++
        }
    }
}
